### ISSUE TITLE
Will not compile due to linker error

### ISSUE THREAD

**Issue by erentar** [Isysxp/Tek-Term#2]
The error is as following:
```
gcc -o tek40xx tek_main.o tek_display.o tek_telnet.o tek_drawline.o font-5x7.o `sdl2-config --libs` -lm
/usr/bin/ld: tek_display.o:~/Tek40xx/tek_video.h:103: multiple definition of `tek_socket'; tek_main.o:~/Tek40xx/tek_video.h:103: first defined here
/usr/bin/ld: tek_drawline.o:~/Tek40xx/tek_video.h:103: multiple definition of `tek_socket'; tek_main.o:~/Tek40xx/tek_video.h:103: first defined here
collect2: error: ld returned 1 exit status
make: *** [makefile:12: tek40xx] Error 1
```

note that after this error is given, issuing `gcc -o tek40xx tek_main.o tek_display.o tek_telnet.o tek_drawline.o font-5x7.o \`sdl2-config --libs\` -lm -Wl,--allow-multiple-definition` will compile the binary fine.

**Comment by Isysxp** [comment]
Thanks for this ... an interesting error. Quite a bit of noise about this on the web as there are some suggestions that this appeared somewhere between gcc 8.0 and 10.0
Have updated the makefile and checked with gcc 10. Looks fine now.

________________________________
From: erentar ***@***.***>
Sent: 01 January 2023 8:25 AM
To: Isysxp/Tek40xx ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Isysxp/Tek40xx] Will not compile due to linker error (Issue #2)


⚠ External sender. Take care when opening links or attachments. Do not provide your login details.

The error is as following:

gcc -o tek40xx tek_main.o tek_display.o tek_telnet.o tek_drawline.o font-5x7.o `sdl2-config --libs` -lm
/usr/bin/ld: tek_display.o:~/Tek40xx/tek_video.h:103: multiple definition of `tek_socket'; tek_main.o:~/Tek40xx/tek_video.h:103: first defined here
/usr/bin/ld: tek_drawline.o:~/Tek40xx/tek_video.h:103: multiple definition of `tek_socket'; tek_main.o:~/Tek40xx/tek_video.h:103: first defined here
collect2: error: ld returned 1 exit status
make: *** [makefile:12: tek40xx] Error 1


note that after this error is given, issuing gcc -o tek40xx tek_main.o tek_display.o tek_telnet.o tek_drawline.o font-5x7.o sdl2-config --libs -lm -Wl,--allow-multiple-definition will compile the binary fine.

**Comment by erentar** [comment]
It compiles for me too. Thank you for the swift fix!

**Comment by Isysxp** [comment]
No problem. Any further comments most welcome.

Regards, Ian.

Sent from my iPad

On 2 Jan 2023, at 09:23, erentar ***@***.***> wrote:

﻿

⚠ External sender. Take care when opening links or attachments. Do not provide your login details.

Closed #2<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIsysxp%2FTek40xx%2Fissues%2F2&data=05%7C01%7Cian.schofield%40newcastle.ac.uk%7Cc43597bffa7547f1480208daeca30206%7C9c5012c9b61644c2a91766814fbe3e87%7C1%7C0%7C638082482092675564%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=NqsmXCxuSNQz%2Bz5FF7QUxtTr3HTUl0I9yR52V%2BzFes8%3D&reserved=0> as completed.
